packages/@react-spectrum/tree/src/TreeView.tsx

```diff
@@ -0,0 +1,361 @@
+    <TreeItem
+      {...props}
+      className={renderProps => treeRow({
+        ...renderProps,
+        isLink: !!href
+      })}>
+      <TreeItemContent>
+        {({isExpanded, hasChildRows, level, selectionMode, selectionBehavior, isDisabled, isSelected, isFocusVisible}) => (
+          <div className={treeCellGrid({isDisabled})}>
+            {selectionMode !== 'none' && selectionBehavior === 'toggle' && (
+              // TODO: add transition?
+              <Checkbox
+                isEmphasized
+                UNSAFE_className={treeCheckbox()}
+                UNSAFE_style={{paddingInlineEnd: '0px'}}
+                slot="selection" />
+            )}
+            <div style={{gridArea: 'level-padding', marginInlineEnd: `calc(${level - 1} * var(--spectrum-global-dimension-size-200))`}} />
+            {/* TODO: revisit when we do async loading, at the moment hasChildItems will only cause the chevron to be rendered, no aria/data attributes indicating the row's expandability are added */}
+            {(hasChildRows || hasChildItems) && <ExpandableRowChevronMacros isDisabled={isDisabled} isExpanded={isExpanded} />}
+            <SlotProvider
+              slots={{
+                text: {UNSAFE_className: treeContent({isDisabled})},
+                // Note there is also an issue here where these icon props are making into the action menu's icon. Resolved by 8ab0ffb276ff437a65b365c9a3be0323a1b24656
+                // but could crop up later for other components
+                icon: {UNSAFE_className: treeIcon(), size: 'S'},
+                actionButton: {UNSAFE_className: treeActions(), isQuiet: true},
+                actionGroup: {
+                  UNSAFE_className: treeActions(),
+                  isQuiet: true,
+                  density: 'compact',
+                  buttonLabelBehavior: 'hide',
+                  isDisabled,
+                  overflowMode: 'collapse'
+                },
+                actionMenu: {UNSAFE_className: treeActionMenu(), UNSAFE_style: {marginInlineEnd: '.5rem'}, isQuiet: true}
+              }}>
+              {content}
+            </SlotProvider>
+            <div className={treeRowOutline({isFocusVisible, isSelected})} />
+          </div>
+        )}
+      </TreeItemContent>
+      {nestedRows}
+    </TreeItem>
+  );
+};
+
+interface ExpandableRowChevronProps {
+  isExpanded?: boolean,
+  isDisabled?: boolean,
+  isRTL?: boolean
+}
+
+const expandButton = style<ExpandableRowChevronProps>({
+  gridArea: 'expand-button',
+  height: 'full',
+  aspectRatio: 'square',
+  display: 'flex',
+  flexWrap: 'wrap',
+  alignContent: 'center',
+  justifyContent: 'center',
+  outlineStyle: 'none',
+  transform: {
+    isExpanded: {
+      default: 'rotate(90deg)',
+      isRTL: 'rotate(-90deg)'
+    }
+  },
+  transition: '[transform ease var(--spectrum-global-animation-duration-100)]'
+});
+
+function ExpandableRowChevronMacros(props: ExpandableRowChevronProps) {
```

Comment:
why include "Macros" in the name? is it different than the style macros or is there a non-macros one? otherwise I think it's unnecessary. i know it's internal only, but i started looking for things it might mean to warrant it

Reply:
I don't actually remember why I named it this haha, I think it was just to set it apart from the TableView's ExpandableRowChevron but I'll just rename this